awscli/customizations/history/list.py

```diff
@@ -0,0 +1,499 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import datetime
+
+import colorama
+
+from botocore.compat import OrderedDict
+
+from awscli.compat import default_pager
+from awscli.customizations.history.commands import HistorySubcommand
+
+
+class ListCommand(HistorySubcommand):
+    NAME = 'list'
+    DESCRIPTION = (
+        'Shows a list of previously run commands and their command_ids. '
+        'Each row shows only a bare minimum of details including the '
+        'command_id, date, arguments and return code. You can use the '
+        '``history show`` with the command_id to see more details about '
+        'a particular entry.'
+    )
+    _COL_WIDTHS = OrderedDict([
+        ('Command Id', 38),
+        ('Time', 24),
+        ('Arguments', 50),
+        ('Return Code', 13)
+    ])
+
+    def _run_main(self, parsed_args, parsed_globals):
+        self._connect_to_history_db()
+        raw_records = self._db_reader.iter_all_records()
+        records = RecordAdapter(raw_records)
+        if not records.has_next():
+            raise RuntimeError('No recorded history.')
+
+        preferred_pager = self._get_preferred_pager()
+        with self._get_output_stream(preferred_pager) as output_stream:
+            use_color = self._should_use_color(parsed_globals)
+            styler = TableStyler(output_stream, colorize=use_color)
+            formatter = TableFormatter(self._COL_WIDTHS, styler)
+            formatter('history list', records)
+        return 0
+
+    def _get_preferred_pager(self):
```

Comment:
You're going to consolidate with what @kyleknap has for figuring out paging/color options once that's all merged correct?

Reply:
Yeah. I think I would prefer to make the change (if any) in this PR related to the `PAGER` env var instead of mine because this PR consolidates them.

Reply:
Yes.